xvec/accessor.py

```diff
@@ -918,6 +929,128 @@ def to_geodataframe(
         )
         return df
 
+    def agg_geom(self, geom, trans,var,  stat='mean'):
+
+        #Create a GeoSeries from the geometry
+        geo_series = gpd.GeoSeries(geom)
+
+        # Convert the GeoSeries to a GeometryArray
+        geometry_array = geo_series.geometry.array
+
+        xar_chunk = self._obj[var]
+        mask = rasterio.features.geometry_mask(geometry_array, out_shape=(xar_chunk.shape[0], xar_chunk.shape[1]), transform= trans)
+        masked_data = xar_chunk * mask[:, :, np.newaxis]
+        del mask, xar_chunk; gc.collect()
+
+        if stat == 'sum':
+            stat_within_polygons = da.sum(masked_data, axis=(0, 1))
+        elif stat == 'mean':
+            stat_within_polygons = da.mean(masked_data, axis=(0, 1))
+        elif stat == 'median':
+            stat_within_polygons = da.median(masked_data, axis=(0, 1))
+        elif stat == 'max':
+            stat_within_polygons = da.max(masked_data, axis=(0, 1))
+        elif stat == 'min':
+            stat_within_polygons = da.min(masked_data, axis=(0, 1))
+
+        result = stat_within_polygons.compute()
+        del masked_data, stat_within_polygons; gc.collect()
+
+        return result 
+    
+
+
+    def spatial_agg(self, geometries, stat='mean', chunk_size = 2):
+        transform = self._obj.rio.transform()
+        num_cores = multiprocessing.cpu_count() 
```

Comment:
Shall we expose this as `n_jobs` defaulting to `-1` (all threads)?

Reply:
I made it an optional user input with -1 as default.